tests/unit/unit1650.c

```diff
@@ -198,9 +198,8 @@ UNITTEST_START
       struct dohaddr *a;
       a = &d.addr[u];
       if(resp[i].type == DNS_TYPE_A) {
-        snprintf(ptr, len, "%d.%d.%d.%d ",
-                 a->ip.v4 & 0xff, (a->ip.v4>>8) & 0xff,
-                 (a->ip.v4>>16) & 0xff, a->ip.v4 >>24);
+        char *p = (char *)&a->ip.v4;
```

Comment:
Should be "unsigned char *" to allow for platforms where char is by default signed.  Though would only matter if the test were changed to use an address other than 127.0.0.1

Reply:
But a top-bit-set char gets promoted to negative int in the function call.  E.g. on Solaris/SPARC, the program
```
#include <stdio.h>
int main(void)
{
  unsigned ip = 0xc0a80001;
  char *p = (char *)&ip;
  printf("%u.%u.%u.%u\n", p[0], p[1], p[2], p[3]);
  return 0;
}
```
yields the output
`4294967232.4294967208.0.1`
(Sorry if wasting time on an only theoretical nit.)